client/landing/gutenboarding/components/header/index.tsx

```diff
@@ -49,7 +86,7 @@ const Header: FunctionComponent< Props > = ( {
 					<div className="gutenboarding__site-title">
 						{ siteTitle ? siteTitle : NO__( 'Create your site' ) }
 					</div>
-					<DomainPickerButton />
+					<DomainPickerButton>{ domainText ?? NO__( 'Choose a domain' ) }</DomainPickerButton>
```

Comment:
Don't show if we don't have enough information.

```suggestion
					{ domainText && <DomainPickerButton>{ domainText }</DomainPickerButton> }
```